osquery/tables/system/windows/registry.h

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <WbemIdl.h>
+#include <Windows.h>
+#include <intrin.h>
+#include <iostream>
+#include <osquery/tables.h>
+#include <string>
+#include <vector>
+
+namespace osquery {
+namespace tables {
+
+// Windows registry helper macros for key and path sizes
+#define MAX_KEY_LENGTH 255
+#define MAX_VALUE_NAME 16383
+
+/// Takes a Windows FILETIME object and returns seconds since epoch
+static LONGLONG filetimeToUnixtime(FILETIME ft) {
+  LARGE_INTEGER date, adjust;
+  date.HighPart = ft.dwHighDateTime;
+  date.LowPart = ft.dwLowDateTime;
+  adjust.QuadPart = 11644473600000 * 10000;
+  date.QuadPart -= adjust.QuadPart;
+  return date.QuadPart / 10000000;
+};
+
+/// Microsoft helper function for getting the contents of a registry key
+std::vector<std::map<std::string, std::string>> queryKey(HKEY hKey) {
```

Comment:
The:
```cpp
std::vector<std::map<std::string, std::string>>
```
type is actually a `QueryData`.

There's a bit of unneeded construct/copy/move going on between this method and the caller. The caller seems to be doing "some" of the `Row` work, perhaps that can be refactored into this method by supplying the missing columns as arguments (const references would imply minimal added work). Then pass the `QueryData &results` as a non-const reference.

In this method you can make a `Row` and, when finished, add it directly to that `results`. The method becomes a `void` or a `Status` based on your preference, and you'll see at least 2x performance improvement.